src/go/rpk/pkg/cli/cmd/debug/bundle.go

```diff
@@ -0,0 +1,842 @@
+				ReplFactor:    ds.ReplicationFactor,
+			})
+		}
+		cgs2proto, err := admin.ListConsumerGroups()
+		if err != nil {
+			return fmt.Errorf("couldn't list consumer groups: %w", err)
+		}
+		cgNames := make([]string, 0, len(cgs2proto))
+		for name := range cgs2proto {
+			cgNames = append(cgNames, name)
+		}
+		cgDescs, err := admin.DescribeConsumerGroups(cgNames)
+		if err != nil {
+			if len(cgDescs) == 0 {
+				return fmt.Errorf("couldn't list describe consumer groups: %w", err)
+			}
+		}
+		cgs := make([]*consumerGroup, 0, len(cgDescs))
+		for _, desc := range cgDescs {
+			d := desc
+			members := make([]*consumerGroupMember, 0, len(d.Members))
+			for _, m := range d.Members {
+				members = append(members, &consumerGroupMember{
+					Host: m.ClientHost,
+					Id:   m.ClientId,
+				})
+			}
+			cgs = append(cgs, &consumerGroup{
+				Id:      d.GroupId,
+				Members: members,
+			})
+		}
+		bytes, err := json.Marshal(&kafkaMetadata{
+			Brokers:        brokers,
+			Topics:         topics,
+			ConsumerGroups: cgs,
+		})
+		if err != nil {
+			return fmt.Errorf("couldn't marshal Kafka metadata: %w", err)
+		}
+		return writeFileToZip(ps, "kafka.json", bytes)
+	}
+}
+
+// Walks the redpanda data directory recursively, and saves to the bundle
+// a JSON map where the keys are the file/ dir paths, and the values are
+// objects containing their data: size, mode, the file or dir it points to
+// if the current file is a symlink, the time it was modified, its owner and
+// its group, as well as an error message if reading that specific file failed.
+func saveDataDirStructure(ps *stepParams, conf *config.Config) step {
+	type fileInfo struct {
+		Size     string    `json:"size"`
+		Mode     string    `json:"mode"`
+		Symlink  string    `json:"symlink,omitempty"`
+		Error    string    `json:"error,omitempty"`
+		Modified time.Time `json:"modified"`
+		User     string    `json:"user"`
+		Group    string    `json:"group"`
+	}
+	var walk func(string, map[string]*fileInfo) error
+	walk = func(root string, files map[string]*fileInfo) error {
+		return filepath.WalkDir(
+			root,
+			func(path string, d fs.DirEntry, readErr error) error {
+				// Prevent infinite loops.
+				if _, exists := files[path]; exists {
+					return nil
+				}
+
+				files[path] = &fileInfo{}
+
+				info, err := d.Info()
+				if err != nil {
+					files[path].Error = err.Error()
+					// If reading a directory failed, then skip it altogether.
+					if d.IsDir() {
+						return fs.SkipDir
+					}
+					// If it's just a file, just return and move to the
+					// next entry.
+					return nil
+				}
+
+				files[path].Size = units.HumanSize(float64(info.Size()))
+				files[path].Mode = info.Mode().String()
+				files[path].Modified = info.ModTime()
+
+				// The user and group are only available through the
+				// underlying syscall object.
+				sys, ok := info.Sys().(*syscall.Stat_t)
+				if ok && sys != nil {
+					u, err := user.LookupId(fmt.Sprint(sys.Uid))
```

Comment:
I think `strconv.Itoa` is more common for this type of thing.

Reply:
Uid is a uint32, so I went for `fmt.Sprint` instead.